extensions/common/context.cc

```diff
@@ -42,12 +43,37 @@ using Envoy::Extensions::Common::Wasm::Null::Plugin::getValue;
 namespace Wasm {
 namespace Common {
 
+const char kRbacFilterName[] = "envoy.filters.http.rbac";
+const char kRbacPermissivePolicyIDField[] = "shadow_effective_policy_id";
+const char kRbacPermissiveEngineResultField[] = "shadow_engine_result";
+
+namespace {
+
+// Extract fqdn from Istio cluster name. If cluster name does not follow Istio
+// convention, fqdn will be left as empty string.
```

Comment:
1. Please add documentation and example of pilot generated cluster names.
2. Also add tests for the same.

Reply:
In addition to cluster name, this is the cluster metadata for TD.
Use it as a 2nd level fall back.

``` c++
getStringValue({"cluster_metadata", "filter_metadata", "com.google.trafficdirector", "backend_service_name"}, &service_name);
```
```yaml
    "filter_metadata":
        "com.google.trafficdirector":
              "backend_service_name": "csm-serve404"
```

Reply:
scrap that. Do not read from cluster metadata yet.

Reply:
Done. Update SD e2e test a bit to reflect this change. It is a bit of hard to add unit test due to usage of wasm API.